### PR TITLE
Refactor modules

### PR DESCRIPTION
**Pull request by vnmabus**
Modules have been refactored to follow a more hierarchical structure.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -56,15 +56,7 @@
               'sphinx.ext.doctest' ]
 
 doctest_global_setup = '''
-import skfda
```
Comment (pablomm) [inline docs/conf.py]: Without these imports I do not pass the local test, there must be a problem in the configuration of doctest on my computer, which does not automatically import the modules of the examples.

File: skfda/exploratory/depth_measures.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
```
Comment (pablomm) [inline skfda/exploratory/depth_measures.py]: I would call this module `depth`.

File: docs/modules/representation.rst [hunk 3]
```diff
@@ -20,7 +20,7 @@ and basic visualisation of a FDataGrid.
 .. autosummary::
    :toctree: autosummary
 
-   skfda.grid.FDataGrid
+   skfda.representation.grid.FDataGrid
```
[the remark this answers is not in the document]
Comment (vnmabus) [reply docs/modules/representation.rst]: I am not against this. Let us see what the others think.

File: skfda/representation/__init__.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+from ._functional_data import FData
+from .basis import FDataBasis
+from .grid import FDataGrid
+
+from . import basis
+from . import extrapolation
+from . import interpolation
+from . import grid
```
[the remark this answers is not in the document]
Comment (vnmabus) [reply skfda/representation/__init__.py]: I think it is better to keep it, for symmetry and possible future additions.

File: skfda/math/_lfd.py [hunk 5]
```diff
@@ -5,7 +5,7 @@
 __email__ = "92manso@gmail.com"
 
 
-class Lfd:
+class LinearDifferentialOperator:
```
[the remark this answers is not in the document]
Comment (vnmabus) [reply skfda/math/_lfd.py]: The `representation`module is for representation of functional data. I do not think this belongs to that category.

File: skfda/__init__.py [hunk 6]
```diff
@@ -23,18 +23,16 @@
...
 
-from .core import *
-from .functional_data import FData
-from .basis import FDataBasis
-from .grid import FDataGrid
-from .math import mean, var, gmean, log, log2, log10, exp, sqrt, \
-    cumsum, inner_product, cov
-from .metrics import lp_distance, norm_lp
+from .representation import FData
+from .representation import FDataBasis
```
[the remark this answers is not in the document]
Comment (vnmabus) [reply skfda/__init__.py]: I prefer to make clear that basis is a representation of functional data. You can always rename it on import.

File: skfda/__init__.py [hunk 7]
```diff
@@ -23,18 +23,16 @@
...
+from .representation import FDataGrid
+from .math import log, log2, log10, exp, sqrt, cumsum, inner_product
+from .math.metrics import lp_distance, norm_lp
```
[the remark this answers is not in the document]
Comment (vnmabus) [reply skfda/__init__.py]: I agree, but let us wait to see the others opinions.

File: skfda/math/_math.py [hunk 8]
```diff
@@ -5,10 +5,8 @@
```
[the remark this answers is not in the document]
Comment (vnmabus) [reply skfda/math/_math.py]: Even if they are the same, I am not sure if we should keep them, if only for not having to import and understand another package.

File: docs/conf.py (hunk 1 above)
Comment (vnmabus) [reply docs/conf.py]: We should check that, because I can run the tests just fine.

File: skfda/exploratory/depth_measures.py (hunk 2 above)
Comment (vnmabus) [reply skfda/exploratory/depth_measures.py]: I though about that. I do not have a strong opinion either way.

**Comment by vnmabus** [comment]
> I add some personal suggestions about the new structure.
> 
> I wouldn't add the `ml` module, because I think the clustering, classification and regression problems are broad enough to be in differentiated modules, and I think there is no advantage to doing `from skfda.ml.clustering import KMeans` instead of `from skfda.clustering import KMeans`.

I think that clustering, classification and regression are related, and are NOT at the same level as exploration and preprocessing. Also, ml is a very short name, so the lenght of the import is not problematic.

File: skfda/exploratory/depth_measures.py (hunk 2 above)
Comment (vnmabus) [reply skfda/exploratory/depth_measures.py]: Done in 0a3284ccbcda2bdd00249a21a57d2db127ef6703.

File: skfda/__init__.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (vnmabus) [reply skfda/__init__.py]: Done in d87260b2a5e9e78d9f53e540ed6439d7fe9d572b.

**Pull request merged**